packages/wallet-apis/src/decorators/smartWalletActions.ts

```diff
@@ -0,0 +1,138 @@
+import type {
+  Hex,
+  WaitForCallsStatusParameters,
+  WaitForCallsStatusReturnType,
+  Address,
+} from "viem";
+import {
+  requestAccount,
+  type RequestAccountParams,
+  type RequestAccountResult,
+} from "../actions/requestAccount.js";
+import {
+  prepareCalls,
+  type PrepareCallsParams,
+  type PrepareCallsResult,
+} from "../actions/prepareCalls.js";
+import {
+  sendPreparedCalls,
+  type SendPreparedCallsParams,
+  type SendPreparedCallsResult,
+} from "../actions/sendPreparedCalls.js";
+import {
+  sendCalls,
+  type SendCallsParams,
+  type SendCallsResult,
+} from "../actions/sendCalls.js";
+import {
+  listAccounts,
+  type ListAccountsParams,
+  type ListAccountsResult,
+} from "../actions/listAccounts.js";
+import {
+  getCallsStatus,
+  type GetCallsStatusParams,
+  type GetCallsStatusResult,
+} from "../actions/getCallsStatus.js";
+import {
+  signSignatureRequest,
+  type SignSignatureRequestParams,
+  type SignSignatureRequestResult,
+} from "../actions/signSignatureRequest.js";
+import {
+  signPreparedCalls,
+  type SignPreparedCallsParams,
+  type SignPreparedCallsResult,
+} from "../actions/signPreparedCalls.js";
+import { signMessage, type SignMessageParams } from "../actions/signMessage.js";
+import {
+  signTypedData,
+  type SignTypedDataParams,
+} from "../actions/signTypedData.js";
+import {
+  grantPermissions,
+  type GrantPermissionsParams,
+  type GrantPermissionsResult,
+} from "../actions/grantPermissions.js";
+import { waitForCallsStatus } from "../actions/waitForCallsStatus.js";
+import type { BaseWalletClient } from "../types.js";
+import { createInternalState } from "../internal.js";
+
+export type SmartWalletActions<
```

Comment:
if the client we are extending is a viem `WalletClient`, we get `getCallsStatus` & `waitForCallsStatus` for free. we don't need to add them on.